### ISSUE TITLE
COCO metrics always returning 0.0

### ISSUE THREAD

**Issue by hugopi** [keras-team/keras-cv#1861]
Hello KerasCV Team,

I hope this message finds you well. I wanted to reach out regarding an issue I am encountering with the COCO metrics in my project. While following the tutorial guidelines, I noticed that the cocoMetrics display a value of 0.0 during both training and evaluation, despite achieving good results. The only time I managed to obtain some values was when I set a large batch size.

(Based on visual inspection, it seems that my model has a poor recall as it is missing a significant number of true positives.)

I wanted to inquire whether there are any parameters that can be adjusted to obtain a correct confusion matrix, similar to the one in the decoder. (IOU_threshold, confidence threshold).

Furthermore, I noticed a particular line in the script "keras-cv/keras_cv/metrics/object_detection/box_coco_metrics.py" that caught my attention:

Line 298: `tf.math.reduce_sum(tf.cast(y_pred["classes"] != -1, tf.int32), axis=-1)`

I wanted to confirm if this line refers to the previous method of decoding predictions, where -1 was set when there was no prediction or if it's normal that classes should be != from -1. It might not be directly related to my issue, but I thought it was worth mentioning.

Here is a snippet of the relevant code for reference:

```
trained_model = create_model()
trained_model.load_weights(CHECKPOINT_PATH_INF)

optimizer = tf.keras.optimizers.SGD(
    learning_rate=LEARNING_RATE, momentum=MOMENTUM, global_clipnorm=GLOBAL_CLIP_NORM
)

coco_metrics = keras_cv.metrics.BoxCOCOMetrics(bounding_box_format="xywh", evaluate_freq=1)

model.compile(
    classification_loss=classification_loss,
    box_loss=box_loss,
    optimizer=optimizer,
    weight_decay=WEIGHT_DECAY,
    metrics=[coco_metrics],
    jit_compile=False,
)

coco_metrics.reset_state()
result = trained_model.evaluate(train_ds.take(40), verbose=0)
result = coco_metrics.result(force=True)

kerasCV_utils.print_metrics(result)

```

Here are the results I obtained for different batch sizes:
Results for batch_size = 4

Metrics:
----------------------------
MaP                         : 0.00
MaP@[IoU=50]                : 0.00
MaP@[IoU=75]                : 0.00
MaP@[area=small]            : 0.00
MaP@[area=medium]           : 0.00
MaP@[area=large]            : 0.00
Recall@[max_detections=1]   : 0.00
Recall@[max_detections=10]  : 0.00
Recall@[max_detections=100] : 0.00
Recall@[area=small]         : 0.00
Recall@[area=medium]        : 0.00
Recall@[area=large]         : 0.00

Results for batch_size = 50

Metrics:
----------------------------
MaP                         : 0.01
MaP@[IoU=50]                : 0.02
MaP@[IoU=75]                : 0.00
MaP@[area=small]            : 0.01
MaP@[area=medium]           : 0.00
MaP@[area=large]            : 0.00
Recall@[max_detections=1]   : 0.01
Recall@[max_detections=10]  : 0.01
Recall@[max_detections=100] : 0.01
Recall@[area=small]         : 0.01
Recall@[area=medium]        : 0.00
Recall@[area=large]         : 0.00

example of result : 
![image](https://github.com/keras-team/keras-cv/assets/90212732/69845c27-0591-4db3-bca1-d4ea91aa0154)

**Comment by LukeWood** [comment]
hi HugoPi!


> I wanted to confirm if this line refers to the previous method of decoding predictions, where -1 was set when there was no prediction or if it's normal that classes should be != from -1. It might not be directly related to my issue, but I thought it was worth mentioning.

Correct.   keras_cv.bounding_box.to_dense() pads -1s on to make tensors dense, so if you want you can use a reduce sum like this to count how many detections the model produced.  Thats what I did here. 

It doesn't seem the metrics are always returning 0, it looks like the metrics are returning low scores.  Your predictions really don't look like they'll hit a strong IoU with the GT boxes, so I'm not super surprised that the values are this low.

>  despite achieving good results
Can you elaborate on what good results are?  I think this would help debug the issue.

What specific issue are you concerned about?  happy to answer questions

**Comment by sachinprasadhs** [comment]
Thanks for reporting the issue! We have consolidated the development of KerasCV into the new **KerasHub** package, which supports image, text, and multi-modal models. Please read [the announcement](https://github.com/keras-team/keras-hub/issues/1831). `KerasHub` will support all the core functionality of `KerasCV`.

`KerasHub` can be installed with `!pip install -U keras-hub`. Documentation and guides are available at [keras.io/keras_hub](https://keras.io/keras_hub/).

With our focus shifted to KerasHub, we are not planning any further development or releases in KerasCV. If you encounter a KerasCV feature that is missing from KerasHub, or would like to propose an addition to the library, please file [an issue with KerasHub](https://github.com/keras-team/keras-hub/issues).
